src/routes/api/users/getById.js

```diff
@@ -1,16 +1,45 @@
 const logger = require("../../../logger");
+const validator = require("validator");
+const prisma = require("../../../model/data/prismaClient");
 const { createSuccessResponse } = require("../../../response");
 
 /**
  * Get a user by their id
  */
-
-// eslint-disable-next-line no-unused-vars
 module.exports = async (req, res, next) => {
   logger.debug(
     { user: req.user, id: req.params.id },
     `received request: GET /v1/users/:userId`,
   );
 
-  return res.status(418).json(createSuccessResponse());
+  const userId = req.params.userId;
+  if (!validator.isHash(userId, "sha256")) {
+    return next({ status: 400, message: "Invalid hash" });
+  }
+
+  let user;
+
+  try {
+    user = await prisma.user.findUnique({
+      where: { hashedEmail: userId }, //searching prisma for the userId (hashed email)
```

Comment:
nitpick: we could remove this comment. i think this is self-explanatory

```suggestion
      where: { hashedEmail: userId },
```